src/kanata/mod.rs

```diff
@@ -319,6 +324,8 @@ impl Kanata {
             live_reload_requested: false,
             overrides: cfg.overrides,
             override_states: OverrideStates::new(),
+            #[cfg(target_os = "macos")]
+            kbd_in_paths: cfg.items.macos_dev,
```

Comment:
It kinda doesn't make sense to assign `cfg.items.macos_dev` to `kbd_in_paths`, because don't `macos_dev` don't contain device paths but device names. But it's just the naming issue.

And generally, about selecting devices in macos: instead of having `macos-dev`, maybe there should be `macos-dev-names-include` and `macos-dev-names-exclude` options, to keep it similar to linux device cfg options?

Reply:
I agree regarding both renaming `kbd_in_path` and implementing the include and exclude functionality.
I originally wanted to support include and exclude but got stuck sending the keyboards' names from driverkit to kanata.
I'm quite new to rust and couldn't figure out a way to do it.
If you can direct me to a way where I can send a list of strings from c++ to rust that would be awesome!

Reply:
I agree on both accounts.
Supporting include and exclude would require changing driverkit. 
Do you think we can go with only supporting `macos-dev` for now and open another issue to support `dev-names-include` and `macos-dev-names-exclude` ?

Reply:
> Do you think we can go with only supporting macos-dev for now and open another issue to support dev-names-include and macos-dev-names-exclude

We can add support for `macos-dev-names-exclude` later for sure, but the confusion between `macos-dev` and `macos-dev-names-include` should be sorted out right now. 

My suggestion is to rename `macos_dev` to `macos-dev-names-include`. This is because the option currently named as `macos_dev` is more similar to `linux-dev-names-include` (both operate on device names) than to `linux-dev` (one operates on device paths and another on device names)

Reply:
I suppose so.
Honestly I didn't check the implementation of `linux-dev-names-include` and `exclude` at first. 

I just assumed they work on substrings of keyboard names.
I thought for example providing something like `logitech` to `linux-dev-names-include` would intercept all keyboards with `logitech` in their name.

But if that's not the case and they match exact device names then the answer to your question is yes.
macos-dev matches keyboard names and intercepts them.

I am not sure if macos exposes device paths like on linux.
In this case I think it would make more sense to use `macos-dev-names-include` instead of `macos-dev`

My biggest issue is that I am finding it hard passing keyboard names between C++ and Rust.
If you have any pointers regarding this I'd really appreciate it!

Reply:
> My biggest issue is that I am finding it hard passing keyboard names between C++ and Rust. If you have any pointers regarding this I'd really appreciate it!

The way I know to do it it is to use CStrings: https://doc.rust-lang.org/std/ffi/struct.CString.html

Rust would pass an array of CString pointers to C++, (passed as a pointer+len), and C++ should copy these strings, since Rust owns the backing memory and C++ can't guarantee their lifetime.